packages/story-editor/src/components/library/panes/shopping/shoppingPane.js

```diff
@@ -98,59 +188,122 @@ function ShoppingPane(props) {
     insertElement: state.actions.insertElement,
   }));
 
-  const insertProduct = useCallback(() => {
-    insertElement('product', {
-      width: 25,
-      height: 25,
-      product,
-    });
-    setProduct(null);
-  }, [insertElement, product]);
+  const deleteProduct = useCallback(
+    (product) => {
+      const element = pageProducts.find(
+        (item) => item.productId === product.productId
+      );
+      if (element) {
+        deleteElementById({ elementId: element.elementId });
+      }
+    },
+    [deleteElementById, pageProducts]
+  );
+
+  const insertProduct = useCallback(
+    (product) => {
+      insertElement('product', {
+        width: 25,
+        height: 25,
+        product,
+      });
+    },
+    [insertElement]
+  );
 
   if (!isEnabled) {
     return null;
   }
 
-  const dropDownParams = {
-    hasSearch: true,
-    lightMode: true,
-    onChange,
-    getOptionsByQuery: getProductsByQuery,
-    selectedId: product?.productId,
-    dropDownLabel: __('Product', 'web-stories'),
-    placeholder: isLoading ? __('Loading…', 'web-stories') : '',
-    disabled: isLoading ? true : isSaving,
-    primaryOptions: isLoading ? [] : initialOptions,
-    zIndex: 10,
-  };
+  const RenderItemOverride = forwardRef(
```

Comment:
This is clever! Nice idea!

Unfortunately however, this won't give us the flexibility that we need, such as being able to display products even when this dropdown is closed, or being able to open a sorting dropdown next to it (#11253) without closing this dropdown with the products.

I'd recommend trying a more "classic" approach with a simple search form that queries products upon enter and then displays the list of found products below. This is more in line with other panels such as media library or page templates tab.

For keyboard navigation we can use `role="list"` and `role="listitem"` and use ArrowUp/ArrowDown to navigate between items. When on a single product & pressing Tab, the Plus/Delete icon is focused.

This works basically like in the media library or the custom fonts setting.